src/main/java/org/elasticsearch/index/translog/TranslogSnapshot.java

```diff
@@ -42,21 +42,19 @@
      * Create a snapshot of translog file channel. The length parameter should be consistent with totalOperations and point
      * at the end of the last operation in this snapshot.
      */
-    public TranslogSnapshot(List<ChannelSnapshot> orderedTranslogs, ESLogger logger) {
+    public TranslogSnapshot(List<Translog.Snapshot> orderedTranslogs) {
         this.orderedTranslogs = orderedTranslogs;
-        this.logger = logger;
         int ops = 0;
-        for (ChannelSnapshot translog : orderedTranslogs) {
+        for (Translog.Snapshot translog : orderedTranslogs) {
 
             final int tops = translog.estimatedTotalOperations();
             if (tops < 0) {
```

Comment:
Do we really ever have the case where we must handle some legacy xlogs, mixed in with some non-legacy ones?  I guess on upgrade, when the node starts a shard, it will replay all ops from the legacy translog (local recovery)?  But I guess we do not then flush, so new indexing ops will write into new format xlog, and if we crash then, on restart we have the mixed case?

Maybe we can flush after replaying legacy xlog?  Then we never have a mixed case?